pkg/sql/information_schema.go

```diff
@@ -552,6 +588,25 @@ func forEachColumnInTable(
 	return nil
 }
 
+func forEachForeignKeyInTable(
+	table *sqlbase.TableDescriptor,
+	fn func(
+		*sqlbase.IndexDescriptor,
+		*sqlbase.ForeignKeyReference,
+	) error,
+) error {
+	for i := range table.Indexes {
```

Comment:
Hm, it's hard for me to say with confidence without looking at something concrete, but generally, yeah, I think I'd argue in favor of adding more parameterization to a single, common `GetConstraintInfo` and having it return more and more structured data than having a wider proliferation of simpler but overlapping code -- previously enforcement of things like name uniqueness looked up one set of constraints while renaming another, etc which made me really want to centralize the logic that traversed a desc to get *the* set of constraints.